### PR TITLE
chore(workflow): allow control of docker meta tag

### PR DESCRIPTION
**Pull request by code-forger**
This PR allows control over the docker meta tag (such as `latest` `next` etc)

1. Three workflows (Release 3, 4, and 5) now accept a required choice of `latest`, `next`, or `do_not_tag`.
- Although Release 3 does not use this value, it will pass it to Release 4. Since Release 3 is the 'entrypoint' for releasing arbitrary tags, Release 3 must receive this value and pass it on.

2. The `Normalize meta tag` step of Release 3, 4, and 5 will determine the value, in this order:
- If the previous workflow passed in a value, that will be used
- If the user has specified a value, that will be used
- For automatic runs on the main branch, the value will be `latest`
- For automatic runs on any other branch the value will be `do_not_tag`

3. The value is used while tagging and pushing the docker image (present in Release 4 and 5 only)
- If the value is `do_not_tag`, the meta tag will be skipped (the image will still be tagged with the versions as normal)
- if the value is `do_not_tag`, the major version tag will also be skipped. The major version tag acts like a `latest in this major stream` tag, and so if we are not tagging `latest` we should not tag the major either.
- For any other run, the tag will take the value as specified in `Normalize meta tag`

4. The value is passed forward to the next workflow (present in Release 3 and 4 only)

- Since Release 6 has no use for the value, Release 5 does not need to pass it on

## How Has This Been Tested?
Running workflows in a fork of this repo, including reading the logs from `Normalize meta tag` and seeing the tag used at the tag step

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update
- [ ] Security update

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] These changes should be applied to a maintenance branch.
- [ ] This change requires cross browser checks.
- [ ] Performance tests should be ran against the server prior to merging.
- [ ] This change impacts caching for client browsers.
- [ ] This change impacts HTTP headers.
- [ ] This change adds additional environment variable requirements for One App users.
- [ ] I have added the Apache 2.0 license header to any new files created.

### CODE REVIEW

File: .github/workflows/release-step-3_automatic_prepare-docker-tags.yml [hunk 1]
```diff
@@ -5,6 +5,14 @@ on:
     tag_version:
      description: 'Semantic Version'
      required: true
+    docker_meta_tag:
+     type: choice
+     options:
+      - do_not_tag
+      - latest
+      - next
```
Comment (JAdshead) [inline .github/workflows/release-step-3_automatic_prepare-docker-tags.yml:13]: what about setting a default ?

**Review by JAdshead — APPROVED** [review]

File: .github/workflows/release-step-3_automatic_prepare-docker-tags.yml (hunk 1 above)
Comment (code-forger) [reply .github/workflows/release-step-3_automatic_prepare-docker-tags.yml:13]: it defaults to the first choice:
image

File: .github/workflows/release-step-3_automatic_prepare-docker-tags.yml [hunk 2]
```diff
@@ -30,6 +38,24 @@ jobs:
           echo ::set-output name=tagversion::${{ github.event.inputs.tag_version }}
           fi
           echo $tagversion
+      - name: Normalize meta tag
+        id: set_docker_meta_tag
+        run: |
+          # if a meta tag was specified this was a menual build
```
Comment (Matthew-Mallimo) [inline .github/workflows/release-step-3_automatic_prepare-docker-tags.yml:44]:
```suggestion
          # if a meta tag was specified this was a manual build
```

File: .github/workflows/release-step-4_automatic_docker-prod-build-and-publish.yml [hunk 3]
```diff
@@ -36,6 +44,29 @@ jobs:
...
           fi
+      - name: Normalize meta tag
+        id: set_docker_meta_tag
+        run: |
+          # Always use the value from the last step if it is present
+          if [[ "${{ github.event.client_payload.docker_meta_tag }}" != "" ]]
+          then
+          echo ::set-output name=metatag::${{ github.event.client_payload.docker_meta_tag }}
+          echo From previous step: ${{ github.event.client_payload.docker_meta_tag }}
+          # if a meta tag was specified this was a menual build
```
Comment (Matthew-Mallimo) [inline .github/workflows/release-step-4_automatic_docker-prod-build-and-publish.yml:55]:
```suggestion
          # if a meta tag was specified this was a manual build
```

File: .github/workflows/release-step-5_automatic_docker-dev-build-and-publish.yml [hunk 4]
```diff
@@ -28,6 +36,29 @@ jobs:
...
           fi
+      - name: Normalize meta tag
+        id: set_docker_meta_tag
+        run: |
+          # Always use the value from the last step if it is present
+          if [[ "${{ github.event.client_payload.docker_meta_tag }}" != "" ]]
+          then
+          echo ::set-output name=metatag::${{ github.event.client_payload.docker_meta_tag }}
+          echo From previous step: ${{ github.event.client_payload.docker_meta_tag }}
+          # if a meta tag was specified this was a menual build
```
Comment (Matthew-Mallimo) [inline .github/workflows/release-step-5_automatic_docker-dev-build-and-publish.yml:47]:
```suggestion
          # if a meta tag was specified this was a manual build
```

**Review by 10xLaCroixDrinker — APPROVED** [review]

**Pull request merged**